.github/workflows/e2e.yaml

```diff
@@ -45,7 +46,15 @@ jobs:
                 working-directory: ${{ matrix.directory }}
 
             # run e2e test
-            -
-                run: php ../e2eTestRunner.php
+            -   run: php ../e2eTestRunner.php -c custom/config/rector.php
+                working-directory: ${{ matrix.directory }}
+                if: ${{ matrix.directory == 'e2e/parallel-custom-config' }}
+
+            -   run: php ../e2eTestRunner.php --config custom/config/rector.php
+                working-directory: ${{ matrix.directory }}
+                if: ${{ matrix.directory == 'e2e/parallel-custom-config' }}
+
+            -   run: php ../e2eTestRunner.php
                 working-directory: ${{ matrix.directory }}
+                if: ${{ matrix.directory != 'e2e/parallel-custom-config' }}
 
```

Comment:
We could put the "if" right into the matrix:

```yaml
   matrix:
                php_version: ['8.1']
                actions:
                     -
                          directory: e2e/template-extends
                          runner: php ../e2eTestRunner.php -c custom/config/rector.php
```

Reply:
I'm merging this PR to keep original fix and this syntax improvement separated :+1: